platform/engineering/design-docs/2020-04-03-separate-content-build.md

```diff
@@ -0,0 +1,486 @@
+hash.
+
+Once we split out the content build, however, it's infeasible to continue this
+approach. Instead, we'll rely on the (`Cache-Control`
+header)[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control].
+Setting it to `no-cache` will require browsers to check with the server for
+every request before using the cached assets.
+
+**Note:** This will cause higher traffic for S3 and may result in a lower
+performance for users, as the browser will have to make a request to the server
+before the cache can be used.
+
+### Code Location
+To start, the **content build** will live at
+`vets-website/src/stages/build-content/`. (This should be mostly just a rename
+of `build/` to `build-content/` for clarity.) Once the implementation is
+complete, this code will **move to a new repository.**
+
+The **application build** will consist of **Webpack configuration files** found
+in the **root of the project.** If we need any script files to aid in the build
+process, they will be located in `vets-website/scripts/build/`.
+
+### Testing Plan
+
+#### Consolidated build script
+After the content and application builds have their own build scripts within
+`vets-website`, we'll write a script to run both builds together to reproduce
+the current build script's output. To test that this is working properly we can:
+
+1. Run the script pre-separation (checkout `master` for this)
+1. Rename the `vets-website/build/localhost/` directory
+1. Run the post-separation build script
+1. Check that the all same files are there
+    - We can use `find` and `cmp` for this as needed
+    
+### Logging
+**Question:** What _should_ it log? Anything? Is this a Jenkins / Nomad /
+CircleCI thing?
+
+### Debugging
+
+#### Content
+
+##### Build
+When the [`process-cms-exports`](https://github.com/department-of-veterans-affairs/vets-website/blob/006185b7c40fe85a8b219ae99f3aefb3d8bf0e09/src/site/stages/build/process-cms-exports/index.js)
+system is live, there will be automated tests to validate input from the CMS and
+output from the content transformers.
+
+##### Deploy
+To determine whether a page exists in the drupal content, we can go to
+`staging.va.gov/drupal/debug/`. This page is not available in production,
+however.
+
+#### Application
+
+##### Build
+We have automated unit and end-to-end tests that are run on every CI build.
+
+##### Deploy
+**Question:** How can we tell whether the application assets are live in
+production?  
+**Answer:** Look at www.va.gov/applications/BUILD.txt and compare the `ref` to
+the commit hash you expect to be in production.
+
+### Caveats
+- The content validation doesn't happen in the build job
+  - The application CI runs tests, but the content CI does not
+  - This is because it takes a long time and content editors are used to seeing
+    their changes in production quickly
+  - Validation instead happens in a separate job
+
+### Security Concerns
+There are no new security concerns with a separated build process.
```

Comment:
some ideas for this section: 

- We haven't resolved an [open question](https://github.com/department-of-veterans-affairs/va.gov-team/pull/6138#pullrequestreview-369232868) about bot tokens in circle yet. 
- Circle runs outside of the VA network
- the website build relies on public npm (existing concern)

Reply:
Should I be documenting existing concerns or new concerns that this process introduces?